civicaseextras.civix.php

```diff
@@ -495,3 +508,62 @@ function (phpQueryObject $doc) {
       });
   $angular->add($changeSet);
 }
+
+/**
+ * Adds the case duration column and value to the case list table template of civicase.
+ *
+ * @param Manager $angular as provided by the alter angular hook.
+ */
+function _civicaseextras_civicrm_alterAngular__addCaseDurationToCaseList (\Civi\Angular\Manager &$angular) {
+  $changeSet = \Civi\Angular\ChangeSet::create('inject_case_duration')
+    ->alterHtml('~/civicase/CaseListTable.html',
+      function (phpQueryObject $doc) {
+        $caseDuration = _civicaseextras_get_caseDurationField();
+
+        _civicaseextras_civicrm_alterAngular__addCaseDurationTableHeader($doc, $caseDuration);
+        _civicaseextras_civicrm_alterAngular__addCaseDurationTableCell($doc, $caseDuration);
+      });
+  $angular->add($changeSet);
+}
+
+/**
+ * Adds the case duration table header to the civicase list table.
+ *
+ * @param phpQueryObject $doc
+ * @param array $caseDuration
+ */
+function _civicaseextras_civicrm_alterAngular__addCaseDurationTableHeader(&$doc, $caseDuration) {
+  $doc->find('.civicase__case-list-table thead tr')
+    ->append('<th
+      ng-show="!viewingCase && headers.length"
+      civicase-case-list-sort-header="custom_' . $caseDuration['id'] . '">
+      ' . $caseDuration['label'] . '
+    </th>');
+}
+
+/**
+ * Adds the case duration table cell to the civicase list table.
+ *
+ * @param phpQueryObject $doc
+ * @param array $caseDuration
+ */
+function _civicaseextras_civicrm_alterAngular__addCaseDurationTableCell(&$doc, $caseDuration) {
+  $doc->find('.civicase__case-list-table tbody tr[ng-repeat="item in cases"]')
```

Comment:
I'm no expert on angular, but this approach seems to be very prone to breaking if any classes / ids are changed. It's not possible to do this inside a template instead? Maybe with [regions](https://docs.civicrm.org/dev/en/latest/framework/region/) or something?

Reply:
Yes, this is an issue, but I don't think there's a better solution. Regions or similar can't be implemented because this is plain HTML that is not parsed by smarty.

Also, adding a placeholder on the main repo (Ex: `<div class="civicase__insert-change-here"></div>`) would be tight coupling the main repo to the extension that will modify it.

I'll notify Shitij so we lock this extension to a specific civicase version to avoid breaking changes.